src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts

```diff
@@ -454,18 +460,130 @@ export class HeaderFilter<T extends Slick.SlickData> {
+    }
+
+    private clearFilterValues(
+        gridFiltersArray: GridColumnMap[],
+        columnId: string,
+    ) {
+        const targetGridFilters = gridFiltersArray.find(
+            (gridFilters) => gridFilters[this.gridId],
+        );
+
+        // Return original array if gridId is not found
+        if (!targetGridFilters) {
+            return gridFiltersArray;
+        }
+
+        // Iterate through each ColumnFilterMap and clear filterValues for the specified columnId
+        for (const columnFilterMap of targetGridFilters[this.gridId]) {
+            if (columnFilterMap[columnId]) {
+                columnFilterMap[columnId] = columnFilterMap[columnId].map(
+                    (filterState) => ({
+                        ...filterState,
+                        filterValues: [],
+                    }),
+                );
+            }
+        }
+
+        return [...gridFiltersArray];
     }
 
-    private updateState(newState: ColumnFilterState) {
-        this.queryResultState.provider.setFilterState(newState);
+    /**
+     * Combines two GridColumnMaps into one, keeping filters separate for different gridIds and removing any duplicate filterValues within the same column id
+     * @param currentFiltersArray
+     * @param newFilters
+     * @param columnId
+     * @returns
+     */
+    private combineFilters(
+        gridFiltersArray: GridColumnMap[],
+        newFilterState: ColumnFilterState,
+        columnId: string,
+    ): GridColumnMap[] {
+        // Find the appropriate GridColumnFilterMap for the gridId
+        let targetGridFilters = gridFiltersArray.find(
+            (gridFilters) => gridFilters[this.gridId],
+        );
+
+        if (!targetGridFilters) {
+            // If no GridColumnFilterMap found for the gridId, create a new one
+            targetGridFilters = { [this.gridId]: [] };
+            gridFiltersArray.push(targetGridFilters);
+        }
+
+        let columnFilterMap = targetGridFilters[this.gridId].find(
+            (map) => map[columnId],
+        );
+
+        if (!columnFilterMap) {
+            // If no existing ColumnFilterMap for this column, create a new one
+            columnFilterMap = { [columnId]: [newFilterState] };
+            targetGridFilters[this.gridId].push(columnFilterMap);
+        } else {
+            // Update the existing column filter state
+            const filterStates = columnFilterMap[columnId];
+            const existingIndex = filterStates.findIndex(
+                (filter) => filter.columnDef === newFilterState.columnDef,
+            );
+
+            if (existingIndex !== -1) {
+                // Replace existing filter state for the column
+                filterStates[existingIndex] = newFilterState;
+            } else {
+                // Add new filter state for this column
+                filterStates.push(newFilterState);
+            }
+        }
+
+        return [...gridFiltersArray];
```

Comment:
why does this need be `[...array]` instead of just `array`?  If there's something weird going on that this works around, add a comment please!